[EDU-2450] BUILD MENU PT functions instances (#338)
aziontech/docs: src/content/docs/pt-br/pages/build-journey/edit-edge-app/edit-functions-instances/functions-instances.mdx

---
title: Como instanciar edge functions em sua aplicação

> description: Instantiate serverless functions directly within your edge application.

Reviewer: ```suggestion
description: Instancie funções serverless diretamente na sua edge application.
```

meta_tags: build
namespace: docs_guides_build_instantiate_edge_functions

> permalink: /documentation/products/guides/build/instantiate-edge-functions/

Reviewer: ```suggestion
permalink: /documentacao/produtos/guias/build/instanciar-edge-functions/
```

menu_namespace: buildMenu
---

> Instancie [funções serverless](/en/documentation/products/edge-application/edge-functions-instances/) diretamente dentro de sua edge application.

Reviewer: ```suggestion
Instancie [funções serverless](/pt-br/documentacao/produtos/edge-application/edge-functions-instances/) diretamente dentro de sua edge application.
```

> :::caution[warning]

Reviewer: ```suggestion
:::caution[Atenção]
```

> Você deve ativar o módulo [Edge Functions](/en/documentation/products/edge-application/edge-functions/) para implementar funções em sua aplicação. Se **Edge Functions** estiver ativado, o tempo de computação e as invocações podem gerar custos relacionados ao uso. Consulte a [página de preços](https://www.azion.com/en/pricing/) para mais informações.

Reviewer: ```suggestion
Você deve ativar o módulo [Edge Functions](/pt-br/documentacao/produtos/edge-application/edge-functions/) para implementar funções em sua aplicação. Se **Edge Functions** estiver ativado, o tempo de computação e as invocações podem gerar custos relacionados ao uso. Consulte a [página de preços](https://www.azion.com/pt-br/precos/) para mais informações.
```

:::

> Este guia usa a função pré-configurada *Azion - Hello World*, uma função global incluída em todas as contas Azion. Você pode repetir este processo usando qualquer função já disponível no [Azion Marketplace](https://marketplace.azion.com/) ou [desenvolver sua própria função](/en/documentation/products/guides/edge-functions/api-builder/)

Reviewer: ```suggestion
Este guia usa a função pré-configurada *Azion - Hello World*, uma função global incluída em todas as contas Azion. Você pode repetir este processo usando qualquer função já disponível no [Azion Marketplace](https://marketplace.azion.com/) ou [desenvolver sua própria função](/pt-br/documentacao/produtos/guias/edge-functions/api-builder/)
```

## Via RTM

Nesta seção, você instanciará uma função para sua aplicação.

1. [Acesse o RTM](https://manager.azion.com).
2. No canto superior esquerdo da página, abra o **Products menu**, representado por três linhas horizontais, e selecione **Edge Application**.
3. Clique na edge application que você deseja configurar. 
4. Ative o módulo **Edge Functions** para habilitar funções.
5. Clique no botão **Save**.
6. Vá para a aba **Functions**.
7. Clique em **Add Function**.
8. Nomeie sua instância de função. Por exemplo: `Hello World function`.
9. Selecione a função **Azion - Hello World**.
10. Clique no botão **Save**.

A página de funções agora lista a nova instância criada. No entanto, esta nova função ainda não está ativa em sua aplicação. Você precisa definir o que acionará a função em sua aplicação. Para invocar a função instanciada no URI `xxxxxxxxxx.map.azionedge.net/hello-world`:

> 11. Ainda na página **Edge Application**, navegue até a aba **Rules Engine**.

Reviewer: ```suggestion
11. Ainda na página do **Edge Application**, navegue até a aba **Rules Engine**.
```

12. Clique no botão **Add Rule** e selecione **Request Phase**.
13. Na seção **Criteria**, selecione a variável `${uri}`.

:::note

> A variável `${uri}` pode já estar selecionada por padrão se você não ativou o módulo [Application Acceleration](/en/documentation/products/edge-application/application-acceleration/).

Reviewer: ```suggestion
A variável `${uri}` pode já estar selecionada por padrão se você não ativou o módulo [Application Acceleration](/pt-br/documentacao/produtos/edge-application/application-acceleration/).
```

:::

14. Como operador de comparação, selecione **is equal**.
15. Como argumento, adicione `/hello-world`.
16. Na seção **Behaviors**, selecione **Run Function** da lista de comportamentos.
17. Selecione a função **Hello World** instanciada.
18. Clique no botão **Save**.

> Para ver sua função em execução, acesse sua aplicação [usando seu domínio](/en/documentation/products/guides/configure-a-domain/) e o URI definido na regra, que está no formato `xxxxxxxxx.map.azionedge.net/hello-world`.

Reviewer: link pt

> Para ver sua função em execução, acesse sua aplicação [usando seu domínio](/en/documentation/products/guides/configure-a-domain/) e o URI definido na regra, que está no formato `xxxxxxxxx.map.azionedge.net/hello-world`.

Author: ```suggestion
Para ver sua função em execução, acesse sua aplicação [usando seu domínio](/pt-br/documentacao/produtos/guias/build/configurar-dominio/) e o URI definido na regra, que está no formato `xxxxxxxxx.map.azionedge.net/hello-world`.
```

## Via API

> 1. Execute a seguinte solicitação `PATCH` em seu terminal, substituindo `[TOKEN VALUE]` por seu [personal token](/en/documentation/products/guides/personal-tokens/) e a variável `<edge_application_id>` por [seu ID de edge application](/en/documentation/products/guides/build/configure-main-settings/) para ativar o módulo [Edge Functions](/en/documentation/products/edge-application/edge-functions/):

Reviewer: ```suggestion
1. Execute a seguinte requisição `PATCH` em seu terminal, substituindo `[TOKEN VALUE]` por seu [personal token](/pt-br/documentacao/produtos/gestao-de-contas/personal-tokens/) e a variável `<edge_application_id>` por [seu ID de edge application](/pt-br/documentation/products/guides/build/definir-configuracoes-principais/) para ativar o módulo [Edge Functions](/pt-br/documentacao/produtos/edge-application/edge-functions/):
```

> 1. Execute a seguinte requisição `PATCH` em seu terminal, substituindo `[TOKEN VALUE]` por seu [personal token](/pt-br/documentacao/produtos/gestao-de-contas/personal-tokens/) e a variável `<edge_application_id>` por [seu ID de edge application](/pt-br/documentation/products/guides/build/definir-configuracoes-principais/) para ativar o módulo [Edge Functions](/pt-br/documentacao/produtos/edge-application/edge-functions/):

Author: ```suggestion
1. Execute a seguinte requisição `PATCH` em seu terminal, substituindo `[TOKEN VALUE]` por seu [personal token](/pt-br/documentacao/produtos/guias/personal-tokens/) e a variável `<edge_application_id>` por [seu ID de edge application](/pt-br/documentation/products/guides/build/definir-configuracoes-principais/) para ativar o módulo [Edge Functions](/pt-br/documentacao/produtos/edge-application/edge-functions/):
```

    "edge_functions": true
}'
```

2. Você receberá uma resposta com o valor atualizado.
3. Execute a seguinte solicitação `GET` para recuperar o `edge_function_id` da função *Hello World*:

```bash
curl --location 'https://api.azionapi.net/edge_functions' \
--header 'Accept: application/json; version=3' \
--header 'Authorization: Token [TOKEN VALUE]'
```

4. Você receberá uma resposta semelhante a esta:

```json
{
  ...
  "results": [
    {
      "id": <edge_function_id>,
      "name": "Azion - Hello World",
      "language": "javascript",
      "code": "\r\nasync function handleRequest(request) {\r\n    return new Response(\"Hello World!\",{status:200})\r\n}\r\n\r\naddEventListener(\"fetch\", event => {\r\n    event.respondWith(handleRequest(event.request))\r\n})\r\n\r\n",
      "json_args": {},
      "function_to_run": "handleRequest",
      "initiator_type": "edge_application",
      "active": true,
      "last_editor": "azionmanager@azion.com",
      "modified": "2021-08-10T17:49:26.224184Z",
      "is_proprietary_code": false,
      "reference_count": 4,
      "version": "-",
      "vendor": "azionmanager@azion.com"
    }
  ]
}
```

5. Copie o valor `<edge_function_id>`.

> 6. Execute a seguinte solicitação `POST`, substituindo o valor do ID da edge function pelo valor que você recebeu na resposta anterior:

Reviewer: ```suggestion
6. Execute a seguinte requisição `POST`, substituindo o valor do ID da edge function pelo valor que você recebeu na resposta anterior:
```

```bash
curl --location 'https://api.azionapi.net/edge_applications/<edge_application_id>/functions_instances' \
--header 'Accept: application/json; version=3' \
--header 'Content-Type: application/json' \
--header 'Authorization: Token [TOKEN VALUE]' \
--data '{
  "name": "Hello World",
  "edge_function_id": <edge_function_id>,
  "args": {}
}'
```

7. Você receberá uma resposta semelhante a esta:

```json
{
    "results": {
        "edge_function_id": <edge_function_id>,
        "name": "Hello World",
        "args": {},
        "id": <edge_function_instance_id>
    }
}
```

  | Chave | Descrição |
  | --- | --- |
  | `edge_function_id` | O ID da edge function. Este não é o mesmo que o ID da função instanciada, que é único para cada instância de função da aplicação. |
  | `name` | Nome da função instanciada. |
  | `args` | Definição dos argumentos necessários para a função. Algumas funções não necessitam de argumentos. |
  | `id` | ID da função instanciada. |
  

> 8. Execute a seguinte solicitação `POST` em seu terminal, substituindo `[TOKEN VALUE]` por seu [personal token](/en/documentation/products/guides/personal-tokens/), a variável `<edge_application_id>` por [seu ID de edge application](/en/documentation/products/guides/build/configure-main-settings/), e a variável `<edge_function_instance_id>` pelo ID da instância da função recebida na resposta anterior:

Reviewer: ```suggestion
8. Execute a seguinte requisição `POST` em seu terminal, substituindo `[TOKEN VALUE]` por seu [personal token](/pt-br/documentacao/produtos/gestao-de-contas/personal-tokens/), a variável `<edge_application_id>` por [seu ID de edge application](/pt-br/documentation/products/guides/build/definir-configuracoes-principais/), e a variável `<edge_function_instance_id>` pelo ID da instância da função recebida na resposta anterior:
```

```bash
curl --location 'https://api.azionapi.net/edge_applications/<edge_application_id>/rules_engine/request/rules' \
--header 'Accept: application/json; version=3' \
--header 'Content-Type: application/json' \
--header 'Authorization: Token [TOKEN VALUE]' \
--data '{
    "name": "Set cache setting - /cache",
    "behaviors": [
        {
            "name": "run_function",
            "target": "<edge_function_instance_id>"
        }
    ],
    "criteria": [
        [
            {
                "variable": "${uri}",
                "operator": "is_equal",
                "conditional": "if",
                "input_value": "/hello-world"
            }
        ]
    ]
}'
```

:::note

> Para mais informações sobre o endpoint de regras, veja o guia [sobre como trabalhar com regras](/en/documentation/products/guides/build/work-with-rules-engine/).

Reviewer: link pt

:::

9. Você receberá uma resposta semelhante a esta, confirmando que sua função foi criada com sucesso:

```json
{
    "results": {
        "id": <rule_id>,
        "name": "Run function - /hello-world",
        "phase": "request",
        "behaviors": [
            {
                "name": "run_function",
                "target": "<edge_function_instance_id>"
            }
        ],
        "criteria": [
            [
                {
                    "variable": "${uri}",
                    "operator": "is_equal",
                    "conditional": "if",
                    "input_value": "/hello-world"
                }
            ]
        ],
        "is_active": true,
        "order": 2,
        "description": ""
    }
}
```

10. Aguarde alguns minutos para que as alterações se propaguem.

> Para ver sua função em execução, acesse sua aplicação [usando seu domínio](/en/documentation/products/guides/configure-a-domain/) e o URI definido na regra, que está no formato `xxxxxxxxx.map.azionedge.net/hello-world`.

Reviewer: link
